File: demo/src/main/java/com/lcm/app/main/MainPresenter.java
package com.lcm.app.main;

import android.app.Activity;
import android.app.Application;
import android.content.Context;

import com.lcm.android.mvp.BaseMvpPresenter;
import com.lcm.app.data.db.DBManager;
import com.lcm.app.data.entity.WelfareBean;
import com.lcm.app.weight.ProgressObserver;
import com.lcm.app.data.entity.HttpBaseResult;


import java.util.List;

import javax.inject.Inject;

import io.reactivex.functions.Consumer;


/**
 * ****************************************************************
 * Author: LCM
 * Date: 2017/5/23 下午12:59
 * Desc:
 * *****************************************************************
 */

public class MainPresenter extends BaseMvpPresenter<MainView> {

    @Inject
    MainModel mainModel;

    @Inject
    DBManager dbManager;


    @Inject
    public MainPresenter() {

    }

    public void load() {
        mainModel.load()
                .subscribe(new ProgressObserver<HttpBaseResult<List<String>>>(getMvpView().getContext()) {
                    @Override
                    public void onNext(HttpBaseResult<List<String>> listHttpBaseResult) {
                        super.onNext(listHttpBaseResult);
                        getMvpView().onLoadSuccess(listHttpBaseResult.getResults().toString());
                    }
                });
    }


    public void getSplash() {
        mainModel.getSplah()
                .subscribe(new Consumer<HttpBaseResult<List<WelfareBean>>>() {
                    @Override
                    public void accept(HttpBaseResult<List<WelfareBean>> listHttpBaseResult) throws Exception {
                        List<WelfareBean> results = listHttpBaseResult.getResults();
                        getMvpView().onLoadSuccess("网络下载数据：："+results.toString());
                        for (WelfareBean w :
                                results) {
//                            dbManager.getWelfareBeanDao().insert(w);
                        }
                    }
                });
    }


    public void readDb() {
//        Query<WelfareBean> build = dbManager.getWelfareBeanDao().queryBuilder().build();
//        List<WelfareBean> list = build.list();
//        getMvpView().onLoadSuccess("从数据库中读取：："+list.toString());
    }
}

File: demo/src/main/java/com/lcm/app/main/MainPresenter.kt
package com.lcm.app.main


import com.lcm.android.mvp.BaseMvpPresenter
import com.lcm.app.data.db.DBManager
import com.lcm.app.weight.ProgressObserver
import com.lcm.app.data.entity.HttpBaseResult

import javax.inject.Inject



/**
 * ****************************************************************
 * Author: LCM
 * Date: 2017/5/23 下午12:59
 * Desc:
 * *****************************************************************
 */

class MainPresenter @Inject
constructor() : BaseMvpPresenter<MainView>() {

    @Inject
    lateinit var mainModel: MainModel

    @Inject
    lateinit var dbManager: DBManager

    fun load() {
        mainModel!!.load()
                .subscribe(object : ProgressObserver<HttpBaseResult<List<String>>>(mvpView.getContext()) {
                    override fun onNext(listHttpBaseResult: HttpBaseResult<List<String>>) {
                        super.onNext(listHttpBaseResult)
                        mvpView.onLoadSuccess(listHttpBaseResult.results.toString())
                    }
                })
    }


    fun getSplash() {
        mainModel!!.splah()
                .subscribe { listHttpBaseResult ->
                    val results = listHttpBaseResult.results
                    mvpView.onLoadSuccess("网络下载数据：：" + results.toString())
                    for (w in results) {
                        //                            dbManager.getWelfareBeanDao().insert(w);
                    }
                }
    }


    fun readDb() {
        //        Query<WelfareBean> build = dbManager.getWelfareBeanDao().queryBuilder().build();
        //        List<WelfareBean> list = build.list();
        //        getMvpView().onLoadSuccess("从数据库中读取：："+list.toString());
    }
}
